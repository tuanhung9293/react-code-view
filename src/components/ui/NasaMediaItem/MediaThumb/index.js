import React from 'react'
import PlayButton from 'components/ui/buttons/PlayButton'
import styles from './MediaThumb.module.css'

class MediaThumb extends React.Component {
  componentDidMount = () => {
    const {source, onFetchCollectionUrl} = this.props
    if (source === 'medias') onFetchCollectionUrl()
  }

  componentDidUpdate = (nextProps) => {
    if (nextProps.itemData.href !== this.props.itemData.href) {
      const {source, onFetchCollectionUrl} = this.props
      if (source === 'medias') onFetchCollectionUrl()
    }
  }

  render() {
    const {itemData, source} = this.props

    return (
      <div className={styles.wrapper}>
        <img src={itemData.thumbUrl} className={styles.thumb} />
        <div className={styles.playButton}>
          <PlayButton to={`/play-media/${source}/${itemData.nasaId}`} />
        </div>
      </div>
    )
  }
}

export default MediaThumb