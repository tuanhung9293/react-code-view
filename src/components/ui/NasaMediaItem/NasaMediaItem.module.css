@value font-sizes: 'styles/font-sizes.css';
@value small, larger from font-sizes;

@value colors: 'styles/colors.css';
@value nasa-dark, nasa-light-dark, nasa-yellow, nasa-gray from colors;


.wrapper {
    position: relative;
    margin-top: 30px;
}

.inner{
    padding-bottom: 24px;
    border-bottom: 4px solid nasa-light-dark;
}

.mediaType {
    position: absolute;
    top: 12px;
    font-weight: 400;
    font-style: normal;
    background-color: nasa-yellow;
}

.mediaType a{
    display: inline-block;
    padding: 4px 8px 4px 12px;
    text-align: center;
    color: white;
    font-size: 10px;
    text-transform: uppercase;
    margin-right: 2px;
    margin-bottom: 2px;
}

.info{
    width: calc(88% - 40px);
    padding: 1px 20px 0 20px;
    background: white;
    left: 6%;
    margin-top: -50px;
    bottom: 0;
    position: relative;
    padding-bottom: 15px;
}

.title{
    font-size: larger;
    margin-top: 20px;
    margin-bottom: 8px;
    color: nasa-dark;
    line-height: 1.3;
}

.meta{
    color: nasa-gray;
    font-size: 11px;
    margin-bottom: 14px;
    text-transform: uppercase;
    display: inline-flex;
}

.postauthor a{
    color: nasa-yellow;
}

.description{
    margin-top: 8px;
    color: nasa-gray;
    line-height: 1.25em;
    font-size: small;
    overflow: hidden;
    text-overflow: ellipsis;
    display: -webkit-box;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
    max-height: 45px;
}

.addButton{
    width: 80%;
    margin: 0 10%;
}
