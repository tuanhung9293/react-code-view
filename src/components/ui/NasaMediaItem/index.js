import React from 'react'
import MediaThumb from './MediaThumb'
import moment from 'moment'
import styles from './NasaMediaItem.module.css'
import clock from 'assets/clock.svg'

function NasaMediaItem({itemData, source, onFetchCollectionUrl, actionComponent: ActionComponent}) {
  return (
    <div className={styles.wrapper}>
      <div className={styles.inner}>
        <MediaThumb
          itemData={itemData}
          source={source}
          onFetchCollectionUrl={onFetchCollectionUrl}
        />
        <div className={styles.mediaType}>
          <a>{itemData.mediaType}</a>
        </div>
        <div className={styles.info}>
          <h4 className={styles.title}>
            <a>{itemData.title}</a>
          </h4>
          <div className={styles.meta}>
            <div className={styles.postauthor}>By <a>{itemData.photographer || 'unknown'}</a></div>
            <div>
              &nbsp;&nbsp;/&nbsp;&nbsp;<img src={clock} height="10px" />
              &nbsp;{moment(new Date(itemData.dateCreated)).format('Do MMMM YYYY')}
            </div>
          </div>
          <div className={styles.description}>{itemData.description}</div>
        </div>

        <div className={styles.addButton}>
          {ActionComponent && <ActionComponent itemData={itemData} />}
        </div>
      </div>
    </div>
  )
}

export default NasaMediaItem
