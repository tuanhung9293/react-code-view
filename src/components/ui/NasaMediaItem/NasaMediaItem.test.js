import React from 'react'
import {shallow} from 'enzyme'
import NasaMediaItem from './'

function createMockComponent() {
  return jest.fn().mockImplementation(props => null)
}
const itemData = {
  mediaType: 'image',
  nasaId: 'nasaId',
  thumbUrl: 'thumbUrl',
  href: 'href-mock-1'
}

it('renders NasaMediaItem', () => {
  expect(shallow(
    <NasaMediaItem 
    itemData={itemData}
    source="medias"
    onFetchCollectionUrl={() => {}}
    actionComponent={createMockComponent()}/>
  )).toMatchSnapshot()
})
