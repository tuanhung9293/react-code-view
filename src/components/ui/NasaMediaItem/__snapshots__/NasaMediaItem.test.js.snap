// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NasaMediaItem 1`] = `
<div
  className="wrapper"
>
  <div
    className="inner"
  >
    <MediaThumb
      itemData={
        Object {
          "href": "href-mock-1",
          "mediaType": "image",
          "nasaId": "nasaId",
          "thumbUrl": "thumbUrl",
        }
      }
      onFetchCollectionUrl={[Function]}
      source="medias"
    />
    <div
      className="mediaType"
    >
      <a>
        image
      </a>
    </div>
    <div
      className="info"
    >
      <h4
        className="title"
      >
        <a />
      </h4>
      <div
        className="meta"
      >
        <div
          className="postauthor"
        >
          By 
          <a>
            unknown
          </a>
        </div>
        <div>
            /  
          <img
            height="10px"
            src="test-file-stub"
          />
           
          Invalid date
        </div>
      </div>
      <div
        className="description"
      />
    </div>
    <div
      className="addButton"
    >
      <mockConstructor
        itemData={
          Object {
            "href": "href-mock-1",
            "mediaType": "image",
            "nasaId": "nasaId",
            "thumbUrl": "thumbUrl",
          }
        }
      />
    </div>
  </div>
</div>
`;
